src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/MetadataTests/DefaultJsonTypeInfoResolverTests.JsonPropertyInfo.cs

```diff
@@ -1019,41 +1019,52 @@ public static void JsonIgnoreConditionIsCorrectlyTranslatedToShouldSerializeDele
 
             static void TestJsonIgnoreConditionDelegate(JsonIgnoreCondition defaultIgnoreCondition, JsonIgnoreCondition? ignoreConditionOnProperty, JsonPropertyInfo property, ModifyJsonIgnore modify)
             {
-                // defaultIgnoreCondition is not taken into accound, we might expect null if defaultIgnoreCondition == ignoreConditionOnProperty
+                // defaultIgnoreCondition is not taken into account, we might expect null if defaultIgnoreCondition == ignoreConditionOnProperty
                 switch (ignoreConditionOnProperty)
                 {
                     case null:
                         Assert.Null(property.ShouldSerialize);
                         break;
+
                     case JsonIgnoreCondition.Always:
                         Assert.NotNull(property.ShouldSerialize);
                         Assert.False(property.ShouldSerialize(null, null));
                         Assert.False(property.ShouldSerialize(null, ""));
                         Assert.False(property.ShouldSerialize(null, "asd"));
+                        Assert.Throws<InvalidCastException>(() => property.ShouldSerialize(null, 0));
 
                         Assert.Null(property.Get);
                         Assert.Null(property.Set);
                         break;
                     case JsonIgnoreCondition.WhenWritingDefault:
-                        Assert.NotNull(property.ShouldSerialize);
-                        Assert.False(property.ShouldSerialize(null, 0));
-                        Assert.True(property.ShouldSerialize(null, 1));
-                        Assert.True(property.ShouldSerialize(null, -1));
+                        TestIntIgnoreWhenWritingDefault();
                         break;
                     case JsonIgnoreCondition.WhenWritingNull:
-                        Assert.NotNull(property.ShouldSerialize);
-                        Assert.False(property.ShouldSerialize(null, null));
-                        Assert.True(property.ShouldSerialize(null, ""));
-                        Assert.True(property.ShouldSerialize(null, "asd"));
+                        TestStringIgnoreWhenWritingDefault();
                         break;
                     case JsonIgnoreCondition.Never:
-                        Assert.NotNull(property.ShouldSerialize);
-                        Assert.True(property.ShouldSerialize(null, null));
-                        Assert.True(property.ShouldSerialize(null, ""));
-                        Assert.True(property.ShouldSerialize(null, "asd"));
+                        Assert.Null(property.ShouldSerialize);
                         break;
```

Comment:
Where is this change coming from? I think JsonIgnoreCondition.Never on property should always translate to `(obj, val) => true` because that should override i.e. `DefaultIgnoreCondition` == `WhenWritingNull`

Reply:
Reverted.